### PR TITLE
Update "tini --version" parsing to be more forgiving of Tini's output format (release build vs git build)

### PR DESCRIPTION
**Pull request by tianon**
**- What I did**

Adjusted the parsing of `tini --version` to account for the different formats of the output (which depend on how `tini` was built):

- `tini version 0.13.0 - git.949e6fa` (Git-based build)
- `tini version 0.13.2` (distribution release build)

Currently, I get periodic lines like the following in my Docker daemon logs:

```
... level=warning msg="failed to retrieve docker-init version: unknown output format: tini version 0.13.2\n"
```

(Which appear to be coming every 20 seconds or so -- sometimes more often, sometimes less often.)

This also fixes `docker info` output to actually show a version:

```console
$ # before
$ docker info | grep '^init'
init version: N/A (expected: 949e6facb77383876aeff8a6944dde66b3089574)

$ # after
$ docker info | grep '^init'
init version: v0.13.2 (expected: 949e6facb77383876aeff8a6944dde66b3089574)
```

See also https://github.com/krallin/tini/issues/79.

**- How I did it**

Updated code in `daemon/info_unix.go`. :innocent:

**- How to verify it**

Running the updated code against `docker-init` and against a distro-provided `tini` that isn't compiled from a Git checkout (they normally use a release tarball instead).

**- Description for the changelog**

- Updated `tini --version` parsing for distro-provided Tini releases

**- A picture of a cute animal (not mandatory but encouraged)**

image

### CODE REVIEW

File: daemon/info_unix.go [hunk 1]
```diff
@@ -56,20 +56,23 @@ func (daemon *Daemon) FillPlatformInfo(v *types.Info, sysInfo *sysinfo.SysInfo)
 
 	v.InitCommit.Expected = dockerversion.InitCommitID
 	if rv, err := exec.Command(DefaultInitBinary, "--version").Output(); err == nil {
+		// examples of how Tini outputs version info:
+		//   "tini version 0.13.0 - git.949e6fa"
+		//   "tini version 0.13.2"
 		parts := strings.Split(strings.TrimSpace(string(rv)), " - ")
```
Comment (thaJeztah) [inline daemon/info_unix.go]: Should we move the parsing to a function and have a unit-test for this? 🤔

Comment (tianon) [inline daemon/info_unix.go]: That'd be an excellent idea IMO!  I can't work on that right away (so if someone else has bandwidth and cares, feel free to take over!), but I'll refactor as soon as I can if someone else doesn't. :smile: :heart:

**Review by vdemeester — APPROVED** [review]
LGTM 🦁

**Review by thaJeztah — APPROVED** [review]
LGTM, I'll have a look later for refactoring, but it's not urgent, so let's merge this as is

**Comment by thaJeztah** [comment]
failures are not related, yolo

**Pull request merged**